internal/charmstore/search_test.go

```diff
@@ -931,3 +958,82 @@ func (s *StoreSearchSuite) TestMultiSeriesCharmSortsSeriesCorrectly(c *gc.C) {
 		newEntity("cs:~charmers/bundle/wordpress-simple-4", 4),
 	})
 }
+
+func (s *StoreSearchSuite) TestOnlyIndexStableCharms(c *gc.C) {
+	ch := storetesting.NewCharm(&charm.Meta{
+		Name: "test",
+	})
+	id := router.MustNewResolvedURL("~test/trusty/test-0", -1)
+	err := s.store.AddCharmWithArchive(id, ch)
+	c.Assert(err, gc.IsNil)
+	err = s.store.SetPerms(&id.URL, "read", "test", params.Everyone)
+	c.Assert(err, gc.IsNil)
+	err = s.store.SetPerms(&id.URL, "development.read", "test", params.Everyone)
+	c.Assert(err, gc.IsNil)
+	err = s.store.SetPerms(&id.URL, "stable.read", "test", params.Everyone)
+	c.Assert(err, gc.IsNil)
+
+	var actual json.RawMessage
+
+	err = s.store.UpdateSearch(id)
+	c.Assert(err, gc.IsNil)
+	err = s.store.ES.GetDocument(s.TestIndex, typeName, s.store.ES.getID(&id.URL), &actual)
+	c.Assert(err, gc.ErrorMatches, "elasticsearch document not found")
+
+	err = s.store.Publish(id, DevelopmentChannel)
+	c.Assert(err, gc.IsNil)
+	err = s.store.UpdateSearch(id)
+	c.Assert(err, gc.IsNil)
+	err = s.store.ES.GetDocument(s.TestIndex, typeName, s.store.ES.getID(&id.URL), &actual)
+	c.Assert(err, gc.ErrorMatches, "elasticsearch document not found")
+
+	err = s.store.Publish(id, StableChannel)
```

Comment:
I see now that this is actually already in place. So why calling `s.store.UpdateSearch(id)` in this test?